redsky/analysis_run_engine.py

```diff
@@ -0,0 +1,175 @@
+import time
+from uuid import uuid4
+
+import traceback
+import subprocess
+
+
+class AnalysisRunEngine:
+    def __init__(self, analysis_db):
+        """Initialize the AnalysisRunEngine
+
+        Parameters
+        ----------
+        analysis_db: databroker.Broker instance
+            The databroker to deposit new headers into
+        """
+        self.an_db = analysis_db
+
+    def __call__(self, hdrs, run_function, *args, md={},
+                 subscription=(), **kwargs):
+        """Run the analysis
+
+        Parameters
+        ----------
+        hdrs: header or list of headers
+            The data to be handed to the RunFunction
+        run_function: RunFunction instance
+            The RunFunction to process the data
+        args: args, optional
+            Arguments to be passed to the RunFunction call
+        md: dict, optional
+            Metadata to be added to the analysis header
+        subscription: iterable of functions, optional
+            Call back function to run on the processed data
+        kwargs: kwargs, optional
+            Key word arguments passed to the RunFunction call
+
+        Returns
+        -------
+
+        """
+        if not isinstance(hdrs, list):
+            hdrs = [hdrs]
+        # issue run start
+        run_start_uid = self.an_db.mds.insert_run_start(
+            uid=str(uuid4()), time=time.time(),
+            parents=[hdr['start']['uid'] for hdr in hdrs],
+            function_name=run_function.__name__,
+            provenance={'args': args,
```

Comment:
Having a hard dep in conda is not all that reasonable, but you can make it pluggable later.